### PR TITLE
DYN-7796 Open Template Locale

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Fixing Templates path locale when Revit is in a different language than English
When Revit Language is French and we open Dynamo Language = Default (inherits Revit Language) if we try to open a template the file explorer location was always en-US. In this fix I'm setting the PreferenceSettings.TemplateFilePath property if the CurrentCulture doesn't match with the locale path used for the templates, also this value will be reflected in DynamoSettings.xml.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Fixing Templates path locale when Revit is in a different language than English

### Reviewers

@QilongTang @reddyashish

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
GIF showing the expected behavior in French.
Revit_BvACGUFDQG

**Comment by RobertGlobant20** [comment]
GIF showing the expected behavior in Spanish.
Revit_8H9O0v32Yg

**Review by reddyashish — APPROVED** [review]

File: src/DynamoCore/Models/DynamoModel.cs [hunk 1]
```diff
@@ -1788,6 +1788,22 @@ private void InitializePreferenceLocations()
             {
                 PreferenceSettings.TemplateFilePath = pathManager.DefaultTemplatesDirectory;
             }
+            var supportedLocales = Configurations.SupportedLocaleDic.Values.ToList<string>();
+
+            //Get the last part of the template path e.f. if the path is C:\ProgramData\Dynamo\Dynamo Core\templates\en-US then currentPathLocale = en-US
+            var pathParts = PreferenceSettings.TemplateFilePath.Split("\\");
+            var currentPathLocale = pathParts[pathParts.Count() - 1];
```
Comment (QilongTang) [inline src/DynamoCore/Models/DynamoModel.cs:1795]: There should be a Last() pointer, no?

Comment (RobertGlobant20) [reply src/DynamoCore/Models/DynamoModel.cs]: Fixed in the next commit: https://github.com/DynamoDS/Dynamo/pull/15657/commits/73fb8229b0af4e2d671467b6d95758dcb6605a67

**Comment by QilongTang** [comment]
@RobertGlobant20 Would you also cherry-pick this?

**Pull request merged**